### ISSUE TITLE
Does example_inputs work in generate_defs_recursive?

### ISSUE THREAD

**Issue by toelli-msft** [microsoft/knossos-ksc#988]
It seems that the same `example_inputs` is provided for *every* function in the call tree? Is that right? If so, is that intentional? I find it rather surprising.

https://github.com/microsoft/knossos-ksc/blob/937d8b760b979ca427e16cf25bec1fbc04a5f100/src/python/ksc/torch_frontend.py#L403-L435

**Comment by awf** [comment]
No, this is broken at the moment.  It works only because float->float functions don't look at their example inputs.  Will be fixed by #vmap
